File: lib/models/cloud_file.dart
/// Represents a file or folder in cloud storage
class CloudFile {
  final String id;
  final String name;
  final String path;
  final bool isDirectory;
  final int? size;
  final DateTime? modifiedTime;
  final String? mimeType;

  CloudFile({
    required this.id,
    required this.name,
    required this.path,
    required this.isDirectory,
    this.size,
    this.modifiedTime,
    this.mimeType,
  });

  /// Check if this is a Scrivener project folder
  bool get isScrivenerProject => isDirectory && name.endsWith('.scriv');

  @override
  String toString() => 'CloudFile(name: $name, isDir: $isDirectory)';
}

File: react-native-writr/src/models/CloudFile.ts
import {CloudFile} from '../types';

export class CloudFileFactory {
  /**
   * Create a cloud file entry
   */
  static create(
    id: string,
    name: string,
    path: string,
    isDirectory: boolean,
    size?: number,
    modifiedTime?: string,
    mimeType?: string,
  ): CloudFile {
    const file: CloudFile = {
      id,
      name,
      path,
      isDirectory,
      size,
      modifiedTime,
      mimeType,
    };

    // Set isScrivenerProject flag
    if (isDirectory && name.endsWith('.scriv')) {
      file.isScrivenerProject = true;
    }

    return file;
  }

  /**
   * Check if this is a Scrivener project folder
   */
  static isScrivenerProject(file: CloudFile): boolean {
    return file.isDirectory && file.name.endsWith('.scriv');
  }

  /**
   * Format file size as human-readable string
   */
  static formatSize(bytes?: number): string {
    if (!bytes) return '';
    if (bytes < 1024) return `${bytes} B`;
    if (bytes < 1024 * 1024) return `${(bytes / 1024).toFixed(1)} KB`;
    return `${(bytes / (1024 * 1024)).toFixed(1)} MB`;
  }
}
